### DISCUSSION TITLE
sqlpage.hash_password() for log in form.

### DISCUSSION THREAD

**Discussion by ArianYN** [sqlpage/SQLPage#635] [Q&A]
I am making custom log in form, and have created a "user" with a password hashed by sqlpage.hash_password()

I am now trying to check if the inputted info from an earlier form is correct:

```sql
SET redirect_link = (
    SELECT
        CASE
            WHEN EXISTS (
                SELECT 1 
                FROM user_authentication 
                WHERE username = :Username 
                AND password = sqlpage.hash_password(:Password)
            ) 
            THEN 'example_page1.sql'
            ELSE 'example_page2.sql'
        END
);

SELECT
    'redirect' as component,
    $redirect_link as link;
```

But this does not work. Does password_hash not work as i imagined, where i can confirm the 2 hashed passwords or is there anything else wrong?

**Comment by lovasoa** [comment]
Indeed, sqlpage.hash_password does not work the way you imagine. It generates [*salted*](https://en.wikipedia.org/wiki/Salt_(cryptography)) hashes, which are more secure, but contain a part of randomness: if you call hash_password twice with the same password, you will get two different results.

You should use the [authentication component](https://sql.datapage.app/component.sql?component=authentication) to check the user's password. Not only is it more secure, it's also easier to use.

And if I make ask: what are you trying to achieve here ? If the only thing behind the authentication is a redirection, then your user can simply load `example_page1.sql` in their browser directly and bypass your authentication.
